File: cdktf-examples/java/src/main/java/com/mycompany/app/MainStack.java
package com.mycompany.app;

import java.io.File;
import com.mycompany.constructs.*;
import software.constructs.Construct;
import com.hashicorp.cdktf.TerraformStack;
import com.hashicorp.cdktf.providers.kubernetes.provider.*;

public class MainStack extends TerraformStack {
        public MainStack(final Construct scope, final String id) {
                super(scope, id);

                new KubernetesProvider(this, "provider", KubernetesProviderConfig.builder()
                                .configPath(new File("../../kubeconfig.yaml").getAbsolutePath())
                                .build());

                new SimpleKubernetesWebApp(this, "app_frontend", new SimpleKubernetesWebAppConfig()
                                .setImage("localhost:5000/nocorp-frontend:latest")
                                .setReplicas(3)
                                .setApp("myapp")
                                .setComponent("frontend")
                                .setEnvironment("dev")
                                .setPort(30001));

        }
}

File: app/MainStack.cs
using System;
using System.IO;
using System.Collections.Generic;
using Constructs;
using HashiCorp.Cdktf;
using HashiCorp.Cdktf.Providers.Kubernetes.Provider;
using HashiCorp.Cdktf.Providers.Kubernetes.Deployment;


namespace MyCompany.MyApp
{
    class MainStack : TerraformStack
    {
        public MainStack(Construct scope, string id) : base(scope, id)
        {
            new KubernetesProvider(this, "k8s", new KubernetesProviderConfig
            {
                ConfigPath = Path.Join(Environment.CurrentDirectory, "../kubeconfig.yaml"),
            });


            new KubernetesWebAppDeployment(this, "deployment", new KubernetesWebAppDeploymentConfig
            {
                Image = "nginx:latest",
                Replicas = 2,
                App = "myapp",
                Component = "frontend",
                Environment = "dev",
                Env = new Dictionary<string, string> { }
            });

            new KubernetesNodePortService(this, "service", new KubernetesNodePortServiceConfig
            {
                Port = 30001,
                App = "myapp",
                Component = "frontend",
                Environment = "dev",
            });
        }
    }
}